rust/src/mime/mime.rs

```diff
@@ -0,0 +1,581 @@
+        match mime_parse_header_token(input) {
+            Ok((rem, t)) => {
+                tokens.insert(t.0, t.1);
+                // should never happen
+                debug_validate_bug_on!(input.len() == rem.len());
+                if input.len() == rem.len() {
+                    //infinite loop
+                    return Err(Err::Error(make_error(input, ErrorKind::Eof)));
+                }
+                input = rem;
+            }
+            Err(_) => {
+                // keep first tokens is error in remaining buffer
+                break;
+            }
+        }
+    }
+    return Ok((input, HeaderTokens { tokens }));
+}
+
+pub fn mime_find_header_token<'a>(
+    header: &'a [u8], token: &[u8], sections_values: &'a mut Vec<u8>,
+) -> Option<&'a [u8]> {
+    match mime_parse_header_tokens(header) {
+        Ok((_rem, t)) => {
+            // in case of multiple sections for the parameter cf RFC2231
+            let mut current_section_slice = Vec::new();
+
+            // look for the specific token
+            match t.tokens.get(token) {
+                // easy nominal case
+                Some(value) => return Some(value),
+                None => {
+                    // check for initial section of a parameter
+                    current_section_slice.extend_from_slice(token);
+                    current_section_slice.extend_from_slice(b"*0");
+                    match t.tokens.get(&current_section_slice[..]) {
+                        Some(value) => {
+                            sections_values.extend_from_slice(value);
+                            let l = current_section_slice.len();
+                            current_section_slice[l - 1] = b'1';
+                        }
+                        None => return None,
+                    }
+                }
+            }
+
+            let mut current_section_seen = 1;
+            // we have at least the initial section
+            // try looping until we do not find anymore a next section
+            loop {
+                match t.tokens.get(&current_section_slice[..]) {
+                    Some(value) => {
+                        sections_values.extend_from_slice(value);
+                        current_section_seen += 1;
+                        let nbdigits = current_section_slice.len() - token.len() - 1;
+                        current_section_slice.truncate(current_section_slice.len() - nbdigits);
+                        current_section_slice
+                            .extend_from_slice(current_section_seen.to_string().as_bytes());
+                    }
+                    None => return Some(sections_values),
+                }
+            }
+        }
+        Err(_) => {
+            return None;
+        }
+    }
+}
+
+pub(crate) const RS_MIME_MAX_TOKEN_LEN: usize = 255;
+
+#[derive(Debug)]
+enum MimeParserState {
+    Start = 0,
+    Header = 1,
+    HeaderEnd = 2,
+    Chunk = 3,
+    BoundaryWaitingForEol = 4,
+}
```

Comment:
nit: do these need values for comparison to C or something? Idtiomatically we'd just use their default and leave off the values.

Reply:
not needed, removing values